js/classes/entities/midground/weapon/sword.js

```diff
@@ -1,224 +1,114 @@
      */
-    static checkType(type) {
-        if (typeof type !== "number" || type % 1 !== 0 || type < 0 || type > 4) {
-            throw new Error("Invalid Sword type: please use a Sword member type (e.g. Sword.TYPE_1).");
-        }
+    isSlicing() {
+        return (Date.now() - this.lastAttack) / 1000 < Sword.DURATION;
     }
 
     /**
-     * Helper method. Gets the value of a property for the current Sword type.
+     * Finds the position of the Sword's bounding box based on Chad's current position, 
+     * scale, and direction.
      * 
-     * @param {string} propertyName the name of the property
-     * @returns the value associated with propertyName for the Sword type
+     * @returns {Vector} the current position of the Sword's bounding box
      */
-    getProperty(propertyName) {
-        return Sword.PROPERTY_TABLE[this.type][propertyName];
-    };
+    calculatePosition() {
+        const chadBBOffset = CHAD.scaleBoundingBoxOffset().x;
+        const offset = (CHAD.facing === "left") ? -this.calculateSize().x + chadBBOffset : chadBBOffset + CHAD.scaledSize.x;
+        return Vector.add(CHAD.pos, new Vector(offset, CHAD.scaleBoundingBoxOffset().y));
+    }
 
     /**
-     * Set the sword to another type.
+     * Calculates the current size of the Sword's bounding box based on Chad's current
+     * scale factor.
      * 
-     * @param {number} type the type of sword we're switching to
+     * @returns {Vector} the size of the Sword's bounding box
      */
-    setType(type) {
-        this.type = type;
-        this.loadAnimations();
+    calculateSize() {
+        return new Vector(Sword.SIZE.x * CHAD.scale.x, Sword.SIZE.y * CHAD.scale.y);
     }
 
     /** Update the Sword. */
     update() {
-        if (CHAD.health <= 0) {
-            return;
-        }
-
-        this.delayTimer -= GAME.clockTick;
+        if (CHAD.health <= 0) return;
 
-        if (GAME.user.jabbing && this.delayTimer <= 0) {
-            // choose from 3 different slash sounds
-            const rand = Math.floor(Math.random() * 3) + 8; //Math.floor(Math.random() * 8) + 1;  use this if you want all 8 sounds
-            const sfx = SFX["SWORD_SWING" + rand];
-            ASSET_MGR.playSFX(sfx.path, sfx.volume);
+        if (!this.isSlicing()) {
+            // the Sword is not currently attacking
 
-            this.isAttacking = true;
-            this.dirX = 1;
-            this.hasHit = false;
+            if (GAME.user.jabbing) { 
+                // a new attack is starting
 
-            this.delayTimer = this.getProperty("DELAY"); // reset the hit delay
-            
-        } else if (this.offsetX >= Sword.MAX_OFFSET) {
-            this.dirX *= -1;
-            this.speed = Sword.RETRACT_SPEED;
-        } else if (this.offsetX <= 0) {
-            this.isAttacking = false;
-        }
+                this.hasHit = false;
+                this.lastAttack = Date.now();
```

Comment:
Why is lastAttack set to the current time?

Reply:
Because `lastAttack` records the time of the most recent attack. Since a new attack is being started here, we need to record it as the most recent attack.